linkis-engineconn-plugins/engineconn-plugins/trino/src/main/scala/org/apache/linkis/engineplugin/trino/creation/TrinoEngineConnFactory.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.engineplugin.trino.creation
+
+import org.apache.commons.lang.StringUtils
+import org.apache.linkis.common.utils.Logging
+import org.apache.linkis.engineconn.common.creation.EngineCreationContext
+import org.apache.linkis.engineconn.common.engineconn.EngineConn
+import org.apache.linkis.engineconn.computation.executor.creation.ComputationSingleExecutorEngineConnFactory
+import org.apache.linkis.engineconn.executor.entity.LabelExecutor
+import org.apache.linkis.engineplugin.trino.conf.TrinoEngineConfiguration
+import org.apache.linkis.engineplugin.trino.entity.TrinoSession
+import org.apache.linkis.engineplugin.trino.exception.TrinoSessionStartFailedException
+import org.apache.linkis.engineplugin.trino.executor.TrinoEngineConnExecutor
+import org.apache.linkis.engineplugin.trino.protocol.GatewayPasswordCallback
+import org.apache.linkis.manager.label.entity.engine.EngineType.EngineType
+import org.apache.linkis.manager.label.entity.engine.RunType.RunType
+import org.apache.linkis.manager.label.entity.engine.{EngineType, RunType}
+
+import java.util.Properties
+import javax.security.auth.callback.PasswordCallback
+
+/**
+ * @author knypys 2022/4/15 13:47
+ */
+class TrinoEngineConnFactory extends ComputationSingleExecutorEngineConnFactory with Logging {
+
+  override protected def newExecutor(id: Int, engineCreationContext: EngineCreationContext, engineConn: EngineConn): LabelExecutor = {
+    engineConn.getEngineConnSession match {
+      case trinoSession: TrinoSession =>
+        new TrinoEngineConnExecutor(id, trinoSession.url, trinoSession.properties, trinoSession.passwordCallback)
```

Comment:
You can refer to the presto engine implementation, which should be able to support tasks to adjust different trino clusters